client/app/stores/message_store.coffee

```diff
@@ -606,13 +606,18 @@ class MessageStore extends Store
         if param.type is 'message' and not _conversationMemoize.size
             messages = _currentMessages
 
-        getMessage = ->
-            index0 = messages.toArray().findIndex (message) ->
-                if param.conversationID isnt 'undefined'
-                    return param.conversationID is message.get 'conversationID'
-                return messageID is message.get 'id'
-            index = param.transform index0
-            messages.get index
+        getMessage = =>
+            _newMessage = (index) ->
```

Comment:
OK, I give up, what is this code suppose to do ?
the `getMessage` function is immediately invoked and not recursive, so seems unnecessary.
the two `index0` will overwrite each other, is that on purpose ?
`messages.toArray().findIndex` -> `message.findEntry`
what does `_newMessage` does exactly ?